### PR TITLE
take a shortcut for classification with only one class in data

### LINKED ISSUE #378: fix ratio 0.0 MLP learning
interactive learning approach with ratio 0.0 is crashing right now

### PR DESCRIPTION
**Pull request by tomsilver**
closes #378

### CODE REVIEW

File: src/torch_models.py [hunk 1]
```diff
@@ -303,6 +305,17 @@ def fit(self, X: Array, y: Array) -> None:
         torch.manual_seed(CFG.seed)
         assert X.ndim == 2
         assert y.ndim == 1
+        # If there is only one class in the data, then there's no point in
+        # learning a NN, since any predictions other than that one class
+        # could only be strange generalization issues.
+        if np.all(y == 0):
+            self._do_single_class_prediction = True
+            self._single_class = False
```
Comment (ronuchit) [inline src/torch_models.py:313]: Can we change this to `predicted_single_class`?

File: src/torch_models.py [hunk 2]
```diff
@@ -337,6 +351,8 @@ def classify(self, x: Array) -> bool:
 
         x is single-dimensional.
         """
+        if self._do_single_class_prediction:
```
Comment (ronuchit) [inline src/torch_models.py:354]: Let's move this into `_classify` so that the asserts later in this method are still run?

File: tests/test_models.py [hunk 3]
```diff
@@ -83,3 +83,19 @@ def test_mlp_classifier():
     model = MLPClassifier(input_size, 10000)
     model.fit(X, y)
     assert time.time() - start_time < 3, "Didn't early stop"
+    # Test with no positive examples.
+    y = np.zeros(len(X))
+    model = MLPClassifier(input_size, 100)
```
Comment (ronuchit) [inline tests/test_models.py:88]: Can we make these 100s bigger, and do a time check like above? (`assert time.time() - start_time < 3, "Didn't early stop"`) The time should be nearly instantaneous for the calls to `fit`.

**Review by ronuchit — COMMENTED** [review]
some suggestions

File: src/torch_models.py (hunk 2 above)
Comment (tomsilver) [reply src/torch_models.py:354]: I want to bypass the normalization, but I refactored so the asserts are still run.

**Review by ronuchit — APPROVED** [review]
lgtm after comments
